### PR TITLE
Added new ErrorType and added customer field to BillpayRequest entity

### PR DESCRIPTION
**Pull request by robyn-mck**
Hi Richard, here are the changes to add the new `CUSTOMER_ACCOUNT_ERROR` ErrorType and the customer field in the BillpayRequest entity.

### CODE REVIEW

File: src/main/java/io/electrum/billpay/model/BillpayRequest.java [hunk 1]
```diff
@@ -49,6 +56,22 @@ public void setSelectedCategory(Category selectedCategory) {
...
+   }
+
+   public Customer getCustomer() {
+      return customer;
+   }
+
+   public void setCustomer(Customer customer) {
+      this.customer = customer;
+   }
+
```
Comment (RichardFourie) [inline src/main/java/io/electrum/billpay/model/BillpayRequest.java:74]: Maybe add an `@since v4.12.0` JavaDoc comment to these new methods.

File: src/main/java/io/electrum/billpay/model/ErrorDetail.java [hunk 2]
```diff
@@ -32,6 +32,7 @@
       TRANSACTION_NOT_SUPPORTED("TRANSACTION_NOT_SUPPORTED"),
       UNABLE_TO_LOCATE_RECORD("UNABLE_TO_LOCATE_RECORD"),
       UNKNOWN_CUSTOMER_ACCOUNT("UNKNOWN_CUSTOMER_ACCOUNT"),
+      CUSTOMER_ACCOUNT_ERROR("CUSTOMER_ACCOUNT_ERROR"),
```
Comment (RichardFourie) [inline src/main/java/io/electrum/billpay/model/ErrorDetail.java:35]: You will need to move this new enum to the end of the enum list since it affects the numeric order of the enums. Also, add a `@since v4.12.0` JavaDoc comment for good measure.

File: pom.xml [hunk 3]
```diff
@@ -4,7 +4,7 @@
   <modelVersion>4.0.0</modelVersion>
   <groupId>io.electrum</groupId>
   <artifactId>billpay-service-interface</artifactId>
-  <version>${major-version}.${minor-version}.${patch-version}</version>
+  <version>${major-version}.${minor-version}.${patch-version}-RC1</version>
```
Comment (RichardFourie) [inline pom.xml:7]: Remember to final version.

File: pom.xml [hunk 4]
```diff
@@ -127,6 +128,11 @@
       <version>${message-masking-sdk-version}</version>
       <scope>provided</scope>
     </dependency>
+    <dependency>
+      <groupId>com.google.code.findbugs</groupId>
+      <artifactId>annotations</artifactId>
+      <version>${google-code-findbugs-version}</version>
+    </dependency>
```
Comment (RichardFourie) [inline pom.xml:135]: Why was this added?

File: src/docs/devguide/hugo/site/content/release-notes.md [hunk 5]
```diff
@@ -1,5 +1,12 @@
 This page describes changes to the Billpay Service Interface implemented across different releases of the interface.
 
+## v4.12.0
+
+Released ? August 2021
```
Comment (RichardFourie) [inline src/docs/devguide/hugo/site/content/release-notes.md:5]: Remember to update the release date.

**Review by RichardFourie — CHANGES_REQUESTED** [review]
A couple of small comments to look at.

File: pom.xml (hunk 4 above)
Comment (robyn-mck) [reply pom.xml:135]: It was required for the Nullable annotation

Comment (RichardFourie) [reply pom.xml:135]: Why not use `javax.validation.constraints.NotNull`?

Comment (RichardFourie) [reply pom.xml:135]: The `@Nullable` annotation is also available under the standard Java library `javax.annotation.Nullable` but we use `javax.validation.constraints.NotNull` to keep things consistent with the rest of the API.

Comment (robyn-mck) [reply pom.xml:135]: Oh - so I should remove the `@Nullable` annotation from the customer field, and it will instead be implied by the fact that it doesn't have a `@NotNull`?

Comment (RichardFourie) [reply pom.xml:135]: It should basically be the same as what is done in the `BillpayResponse` class.

**Review by RichardFourie — APPROVED** [review]
LGTM. Sorry about the stupid comments about the `@Nullable` annotation topic.

File: pom.xml (hunk 4 above)
Comment (RichardFourie) [reply pom.xml]: Sorry, I only realize now reading through the conversation that I was being stupid. I don't know why I suggested using `javax.validation.constraints.NotNull` when the field is supposed to be nullable. You are right in that including or excluding an `@Nullable` annotation has the same effect here.
